### ISSUE TITLE
S1172 reports parameter that is returned in tuple as unused.

### ISSUE THREAD

**Issue by credfeto** [SonarSource/sonar-dotnet#3865]
### Description

S1172 reports that a parameter isn't used when it is returned in a tuple

### Repro steps

`foundItem` is reported as unused in this code:

```
private static (bool success, string? found) Found(string foundItem)
{
   return (success: true, found: foundItem);
}
```

The following code doesn't report the parameter as unused so is inconsistent with the above.

```
private static string? Found2(string foundItem)
{
    return foundItem;
}
```

### Expected behavior

Error not to be triggered.

### Actual behavior

Compiling  returns error:

``Test.cs(20, 60): [S1172] Remove this parameter 'foundItem', whose value is ignored in the method.``

### Known workarounds

None - have to suppress the rule in this place

### Related information

* C#/VB.NET Plugins version : SonarAnalyzer.CSharp 8.15.0.24505
* Visual Studio version:  N/A - using command line to compiue latest rider.
* MSBuild / dotnet version:  dotnet SDK 5.0.101
* SonarScanner for .NET version (if used): n/a
* Operating System: Windows 10

**Comment by pavel-mikula-sonarsource** [comment]
Hi @credfeto ,

Thank you for reporting this case. We've fixed this issue in our latest https://github.com/SonarSource/sonar-dotnet/releases/tag/8.16.0.25740
